libraries/Microsoft.Bot.Builder.Azure/CosmosDbSqlStorage.cs

```diff
@@ -0,0 +1,215 @@
+        public CosmosDbSqlStorage(Uri serviceEndpoint, string authKey, string databaseId, string collectionId, Action<ConnectionPolicy> connectionPolicyConfigurator = null)
+        {
+            if (serviceEndpoint == null)
+            {
+                throw new ArgumentNullException(nameof(serviceEndpoint), "Service EndPoint for CosmosDB is required.");
+            }
+
+            if (string.IsNullOrEmpty(authKey))
+            {
+                throw new ArgumentException("AuthKey for CosmosDB is required.", nameof(authKey));
+            }
+
+            if (string.IsNullOrEmpty(databaseId))
+            {
+                throw new ArgumentException("DatabaseId is required.", nameof(databaseId));
+            }
+
+            if (string.IsNullOrEmpty(collectionId))
+            {
+                throw new ArgumentException("CollectionId is required.", nameof(collectionId));
+            }
+
+            this.databaseId = databaseId;
+            this.collectionId = collectionId;
+
+            var connectionPolicy = new ConnectionPolicy()
+            {
+                UserAgentSuffix = UserAgentSuffix
+            };
+
+            connectionPolicyConfigurator?.Invoke(connectionPolicy);
+
+            this.client = new DocumentClient(serviceEndpoint, authKey, connectionPolicy);
+            this.client.CreateDatabaseIfNotExistsAsync(new Database { Id = databaseId })
+                .Wait();
+
+            this.client.CreateDocumentCollectionIfNotExistsAsync(UriFactory.CreateDatabaseUri(databaseId), new DocumentCollection { Id = collectionId })
+                .Wait();
+        }
+
+        /// <summary>
+        /// Deletes the specified keys from storage.
+        /// </summary>
+        /// <param name="keys">List of keys to remove.</param>
+        /// <returns></returns>
+        public async Task Delete(params string[] keys)
+        {
+            var tasks = keys.Select(key =>
+                this.client.DeleteDocumentAsync(
+                    UriFactory.CreateDocumentUri(this.databaseId, this.collectionId, SanitizeKey(key))));
+
+            await Task.WhenAll(tasks).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Returns an instance of <see cref="StoreItems"/> bag with the values for the specified  keys.
+        /// </summary>
+        /// <param name="keys">List of keys to retrieve.</param>
+        /// <returns>A <see cref="StoreItems"/> instance.</returns>
+        public async Task<StoreItems> Read(params string[] keys)
+        {
+            var storeItems = new StoreItems();
+
+            var parameterSequence = string.Join(",", Enumerable.Range(0, keys.Length).Select(i => $"@id{i}"));
+            var parameterValues = keys.Select((key, ix) => new SqlParameter($"@id{ix}", SanitizeKey(key)));
+            var querySpec = new SqlQuerySpec
+            {
+                QueryText = $"SELECT * FROM c WHERE c.id in ({parameterSequence})",
```

Comment:
My recollection from writing SQL Code years ago is that "Select *" is evil and shouldn't be used in production code as it leads to many unforseen bugs. 

Has this changed in the DocDB world? Is there a way to avoid the "Select *" pattern?